src/Sarif/Visitors/InsertOptionalDataVisitor.cs

```diff
@@ -113,34 +113,36 @@ public override PhysicalLocation VisitPhysicalLocation(PhysicalLocation node)
         public override FileData VisitFileData(FileData node)
         {
             FileLocation fileLocation = node.FileLocation;
+            if (fileLocation != null && _run.OriginalUriBaseIds != null)
```

Comment:
>_run.OriginalUriBaseIds [](start = 40, length = 23)

this isn't right. a file location might be rendered as an absolute file:// URL, which is appropriate for computing hashes or injecting contents.

Reply:
this is another place where we need the key that is associated with the file node, as the file location might be expressed there. how's that full change coming along?

---
In reply to: [221752670](https://github.com/Microsoft/sarif-sdk/pull/1105#discussion_r221752670) [](ancestors = 221752670)